Userland/Libraries/LibJS/Runtime/TypedArray.h

```diff
@@ -184,20 +184,10 @@ class TypedArray : public TypedArrayBase {
 
         // 2. Assert: O is an Integer-Indexed exotic object.
 
-        // NOTE: If the property name is a number type (An implementation-defined optimized
-        // property key type), it can be treated as a string property that has already been
-        // converted successfully into a canonical numeric index.
-
         // 3. If Type(P) is String, then
-        // NOTE: This includes an implementation-defined optimization, see note above!
         if (property_name.is_string() || property_name.is_number()) {
```

Comment:
this is still deviating from the specification, so it needs a comment explaining why it does so, same for below

Reply:
I added back a slightly different comment since we now always call convert.
Is that one ok?

Reply:
yeah that comment looks fine :)